### ISSUE TITLE
Promise rejected before create model overriding defaults: registry.resolver.resolve is not a function

### ISSUE THREAD

**Issue by offirgolan** [emberjs/data#3972]
ember 2.3.0-beta.2+9f7d0aee
ember-data 2.3.0-beta.1+9c55682ba7
ember-cli: 1.13.12

```
Promise rejected before create model overriding defaults: registry.resolver.resolve is not a function@ 23 ms
Source: 	
TypeError: registry.resolver.resolve is not a function
    at resolve (http://localhost:4200/assets/vendor.js:12899:36)
    at Object.Registry.resolve (http://localhost:4200/assets/vendor.js:12439:21)
    at Object.Registry.resolve (http://localhost:4200/assets/vendor.js:12441:33)
    at factoryFor (http://localhost:4200/assets/vendor.js:11974:28)
    at Object.Container.lookupFactory (http://localhost:4200/assets/vendor.js:11861:14)
    at exports.default.TestModule.default.extend.setupModel (http://localhost:4200/assets/test-support.js:2423:38)
    at http://localhost:4200/assets/test-support.js:2566:21
    at Object.initializePromise (http://localhost:4200/assets/vendor.js:63367:7)
    at new Promise (http://localhost:4200/assets/vendor.js:65217:21)
    at nextStep (http://localhost:4200/assets/test-support.js:2565:18)
```

```js
export default DS.Model.extend({
  name: DS.attr('string', { defaultValue: '' }),
  acceptTerms: DS.attr('boolean', { defaultValue: false }),
});
```

```js
import Ember from 'ember';
import {
  moduleForModel,
  test
}
from 'ember-qunit';

moduleForModel('signup', 'Unit | Validations | DS.Model', {

});

test('create model with defaults', function(assert) {
  var model = this.subject();
  // blah blah blah
});
```

**Comment by bmac** [comment]
What version of `ember-qunit` are you using?

**Comment by offirgolan** [comment]
0.4.16

**Comment by offirgolan** [comment]
You can also check out [the build](https://travis-ci.org/offirgolan/ember-cp-validations/jobs/95411594) for more details if it helps at all.

**Comment by bmac** [comment]
cc @rwjblue. 

It looks the the issue here is the `moduleForModel` is attempting to look up the [adapter:application](https://github.com/switchfly/ember-test-helpers/blob/50ada0434696000ffe72a64caf942e6086bd8dad/lib/ember-test-helpers/test-module-for-model.js#L19) which isn't defined for this test. This causes the registry to attempt to use its fallback registry. The error happens because fallback registry has the wrong shape. It looks like [this line](https://github.com/switchfly/ember-test-helpers/blob/b04cc9cff517d3df9f4fde96b8c1839ec10cc91d/lib/ember-test-helpers/test-module.js#L298) from `ember-test-helpers` is overrideing the `resolver` object to be a function. Ember is expecting `resolver` to be an object with a [resolve method](https://github.com/emberjs/ember.js/blob/f0dafd4d5f3e483a8d8111f8f7b75c6dd80f4adb/packages/container/lib/registry.js#L849).

**Comment by rwjblue** [comment]
Confirm, the bug is unrelated to ember-data. It is a bug in ember-test-helpers that we are working on fixing in https://github.com/switchfly/ember-test-helpers/pull/127 (reported in https://github.com/switchfly/ember-test-helpers/pull/126).

**Comment by rwjblue** [comment]
Fix has been released in ember-qunit@0.4.17 and ember-mocha@0.8.8, please update and you should be good to go.

**Comment by offirgolan** [comment]
Thanks @rwjblue :+1:
